src/Lib.P2P/Transports/DatagramStream.cs

```diff
@@ -0,0 +1,144 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.IO;
+using System.Net.Sockets;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Lib.P2P.Transports
+{
+    internal class DatagramStream : Stream
+    {
+        private Socket socket;
+        private bool ownsSocket;
+        private MemoryStream sendBuffer = new MemoryStream();
+        private MemoryStream receiveBuffer = new MemoryStream();
+        private byte[] datagram = new byte[2048];
+
+        public DatagramStream(Socket socket, bool ownsSocket = false)
+        {
+            this.socket = socket;
+            this.ownsSocket = ownsSocket;
+        }
+
+        protected override void Dispose(bool disposing)
+        {
+            if (disposing)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)